mantle/platform/qemu.go

```diff
@@ -306,6 +306,41 @@ func (inst *QemuInstance) SwitchBootOrder() (err error) {
 	return nil
 }
 
+// RemoveBlockDevice deletes the primary device from a qemu instance
+// and sets the seconday device as primary.
+func (inst *QemuInstance) RemoveBlockDevice() error {
+	var primaryDevice string
+	var secondaryDevicePath []string
+	monitor, err := newQMPMonitor(inst.tempdir)
+	if err != nil {
+		return errors.Wrapf(err, "Could not connect to QMP device")
+	}
+	monitor.Connect()
+	defer monitor.Disconnect()
+	blkdevs, err := listQMPBlkDevices(monitor, inst.tempdir)
+	if err != nil {
+		return errors.Wrapf(err, "Could not list block devices through qmp")
+	}
+	for _, dev := range blkdevs.Return {
+		if !dev.Removable {
+			if dev.Inserted.BackingfileDepth == 1 {
+				primaryDevice = dev.DevicePath
+			} else {
+				secondaryDevicePath = append(secondaryDevicePath, dev.DevicePath)
+			}
+		}
+	}
+	err = setBootIndexForDevice(monitor, primaryDevice, -1)
+	if err != nil {
+		return errors.Wrapf(err, "Could not set bootindex for %v", primaryDevice)
+	}
+	err = setBootIndexForDevice(monitor, secondaryDevicePath[0], 1)
```

Comment:
Is there any way we could check the secondary device is also bootable? What would happen if I have a test where it is NOT a mirror and is instead just an empty secondary disk?